src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -2954,3 +3002,41 @@ def execute_query_for_vm(cmd, client, resource_group_name, vm_name, analytics_qu
                        'Please check the status of log analytics workpsace.')
     return client.query(workspace, QueryBody(query=analytics_query, timespan=timespan))
 # endregion
+
+
+# disk encryption set
+def create_disk_encryption_set(cmd, client, resource_group_name, disk_encryption_set_name,
+                               key_url, source_vault, location=None, tags=None, no_wait=False):
+    from msrestazure.tools import resource_id, is_valid_resource_id
+    DiskEncryptionSet, EncryptionSetIdentity, KeyVaultAndKeyReference, SourceVault = cmd.get_models(
+        'DiskEncryptionSet', 'EncryptionSetIdentity', 'KeyVaultAndKeyReference', 'SourceVault')
+    encryption_set_identity = EncryptionSetIdentity(type='SystemAssigned')
+    if not is_valid_resource_id(source_vault):
+        source_vault = resource_id(subscription=client.config.subscription_id, resource_group=resource_group_name,
+                                   namespace='Microsoft.KeyVault', type='vaults', name=source_vault)
+    source_vault = SourceVault(id=source_vault)
+    keyVault_and_key_reference = KeyVaultAndKeyReference(source_vault=source_vault, key_url=key_url)
+    disk_encryption_set = DiskEncryptionSet(location=location, tags=tags, identity=encryption_set_identity,
+                                            active_key=keyVault_and_key_reference)
+    return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, disk_encryption_set_name,
+                       disk_encryption_set)
+
+
+def list_disk_encryption_sets(cmd, client, resource_group_name=None):
+    if resource_group_name:
+        return client.list_by_resource_group(resource_group_name)
+    return client.list()
+
+
+def update_disk_encryption_set(instance, client, resource_group_name, key_url=None, source_vault=None):
+    from msrestazure.tools import resource_id, is_valid_resource_id
+    if not is_valid_resource_id(source_vault):
+        source_vault = resource_id(subscription=client.config.subscription_id, resource_group=resource_group_name,
+                                   namespace='Microsoft.KeyVault', type='vaults', name=source_vault)
+    if key_url:
```

Comment:
user cannot set `key_url` to none. right?

Reply:
It's required.

Reply:
`key_url=None`?

Reply:
When key_url == None, it means do not update key_url. In a disk encryption set, key_url can not be None